python/singa/tensor.py

```diff
@@ -232,14 +232,18 @@ def l1(self):
         '''
         return self.data.L1()
 
-    def set_value(self, x):
+    def set_value(self, x, inplace=True):
         '''Set all elements of the tensor to be the give value.
 
         Args:
             x (float), a float value to be set to all elements.
         '''
         # assert type(x) == float, 'set value only accepts float input'
         # if isinstance(x, float):
+        if not inplace:
+            # return new tensor filled with value
+            raise NotImplementedError
+
         self.data.SetFloatValue(float(x))
```

Comment:
return self if inplace is false; otherwise, return the new tensor?

Reply:
thanks for the review, this is fixed

Reply:
reference https://github.com/apache/singa/pull/570/commits/8cc44683abfb21be284a751f501ba31eb4737022